### PR TITLE
Remove generic_scheduler.go and generic_scheduler_test.go

### LINKED ISSUE #101135: Scheduler Code Deep Cleanup: Scheduler, genericScheduler, and SchedulerAlgorithm
#### What would you like to be added:
Refactor the names of `Scheduler`, `SchedulerAlgorithm`, and `genericScheduler` to something like: 
`Scheduler`, `SchedulerAlgorithm`, and`SchedulerAlgorithmImpl`

#### Why is this needed:
I was reading the scheduler framework code and found out very confusing when reading the `type Scheduler struct` and `type genericScheduler struct` ([link](https://github.com/kubernetes/kubernetes/blob/8edda782f30a377afd031d5d7f4d62551785c441/pkg/scheduler/core/generic_scheduler.go#L79)). The name `genericXXXX` sounds very generic at least to myself... and no comments were there. 

In reality, the `genericScheduler` is just an implementation of the `SchedulerAlgorithm` interface above which will act as the algorithm component in the scheduler object. 

Another note is there was a TODO to rename the `SchedulerAlgorithm struct` before. ([Link](https://github.com/kubernetes/kubernetes/blob/8edda782f30a377afd031d5d7f4d62551785c441/pkg/scheduler/core/generic_scheduler.go#L60))

So, I suggest for the better organization of the scheduler code, some code refactorings of names of `Scheduler`, `SchedulerAlgorithm`, and `genericScheduler` is quite necessary.

### PR DESCRIPTION
**Pull request by ruquanzhao**
Signed-off-by: Ruquan Zhao <ruquan.zhao@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #101135 
#### Special notes for your reviewer:

Follow work of #107135 
Remove `generic_scheduler.go` and `generic_schedule_test.go` since `generic_scheduler` is removed

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -50,6 +53,7 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/metrics"
 	"k8s.io/kubernetes/pkg/scheduler/profile"
 	"k8s.io/kubernetes/pkg/scheduler/util"
+	utiltrace "k8s.io/utils/trace"
```
Comment (kerthcet) [inline pkg/scheduler/scheduler.go:56]: nit: it seems `trace` is enough.

**Comment by kerthcet** [comment]
One nit, otherwise looks good to me as it's just a code migration.

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (ruquanzhao) [reply pkg/scheduler/scheduler.go:56]: Thanks !
But I notice that the `"k8s.io/utils/trace"` are imported as `utiltrace` for most situations. So I prefer to keep it.

**Pull request merged**